### PR TITLE
Document protocol level IPFS schemas

### PR DESCRIPTION
**Pull request by DanielVF**
First pass at documenting our protocol level IPFS schemas.

I looked into several "autodocumenting" tools for working with json-schema, and the results were between bad and terrible.

I've gone with a example-and-details-table approach for the schemas, similarish to https://stripe.com/docs/checkout

image

### CODE REVIEW

File: source/includes/_protocol_schemas.mdown [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+|**price** | For currencies, we support `ETH`, with future plans to support any ERC20 token |
+|**unitsTotal** | The number of units available for sale. Currently limited to `1` |
+
+### Optional Fields
+
+| | |
+|----|-----|
+|expiry | A date that this listing should no longer show in search results. This is suggestion and search engines may ignore it.|
+|media | A list of media elements to be shown with the listing. We currently only support **images** on the **ipfs://** protocol.|
+|securityDeposit| **UNSAFE FIELD - this data duplicates, and may not match, the actual blockchain contract amount.** The currency must be `OGN`. |
```
Comment (franckc) [inline source/includes/_protocol_schemas.mdown]: AFAICT we are not populating this field and it is not a copy of a value from the blockchain.
I added this a while back for future use (to store the amount the seller would ask as security deposit). But it was not really thought through. I'm thinking I should remove it from the V1 schema for now. Thoughts ?

File: source/includes/_protocol_schemas.mdown [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+```
+
+### Required Fields:
+
+| | |
+|----|-----|
+|**schemaId** | Must be set to `http://schema.originprotocol.com/listing-withdraw_v1.0.0` |
+
+
+## Notification Schema
```
Comment (franckc) [inline source/includes/_protocol_schemas.mdown]: I don't think notifications are stored in IPFS.
Also it does not look like the schema under src/IpsInterface/schema/notification.json is actually used at all. I can't find any code referencing it to do any sort of validation. I should probably delete it ?

Maybe we should document this under the "models" which are objects returned as output of Origin-js methods ?
The notification model is [here](https://github.com/OriginProtocol/origin-js/blob/master/src/models/notification.js).

File: source/includes/_protocol_schemas.mdown [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    text: "Buyer handled the notary process very quickly."
+}
+```
+
+### Required Fields:
+
+| | |
+|----|-----|
+|**schemaId** | Must be set to `http://schema.originprotocol.com/review_v1.0.0` |
+|**rating** | A number between 1 and 5. May be either an iteger or a decimal. |
```
[the remark this answers is not in the document]
Comment (DanielVF) [reply source/includes/_protocol_schemas.mdown]: Integer is probably better.

File: source/includes/_protocol_schemas.mdown (hunk 2 above)
Comment (DanielVF) [reply source/includes/_protocol_schemas.mdown]: I've removed this from the docs.

**Comment by franckc** [comment]
Thanks @DanielVF for all your work on this ! Should I go ahead and merge this PR ? Lmk...

**Comment by DanielVF** [comment]
Yup! Merge away!

**Comment by franckc** [comment]
Sorry for the looong delay merging this in ! Thanks for making our docs better :)

**Pull request merged**